### PR TITLE
adding usage charge resource

### PR DESCRIPTION
**Pull request by ch33sybr3ad**
## Description
- adds usage charge resource to shopify api

https://github.com/Shopify/shopify/pull/56192/files

@Shopify/finance @dominiquesr @Not-Jake

### CODE REVIEW

File: lib/shopify_api/resources/recurring_application_charge.rb [hunk 1]
```diff
@@ -19,5 +19,9 @@ def cancel
     def activate
       load_attributes_from_response(post(:activate))
     end
+
+    def customize(capped_amount)
```
Comment (garymardell) [inline lib/shopify_api/resources/recurring_application_charge.rb]: Should this be a hash so its `customize(capped_amount: 10)` ?

File: test/fixtures/usage_charges.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      "billing_on": "2015-07-08",
+      "balance_remaining": "82.00",
+      "balance_used": "18.00"
+
+    },
+    {
+      "id": 359376006,
+      "description": "1000 emails",
+      "price": "1.00",
+      "recurring_application_charge_id": 654381177,
```
Comment (garymardell) [inline test/fixtures/usage_charges.json]: If these are for the same recurring charge they should have the same `balance_remaining`?

**Comment by ch33sybr3ad** [comment]
@Shopify/finance  :eyes: please

Need to merge, so I can write some api unit tests for usage charges within shopify to help @joshubrown with documentation

File: lib/shopify_api/resources/usage_charge.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module ShopifyAPI
+  class UsageCharge < Base
+    init_prefix :recurring_application_charge
+
+    class << self
+      def index(recurring_application_charge_id)
+        UsageCharge.find(:all, :params => { :recurring_application_charge_id => recurring_application_charge_id })
+      rescue ActiveResource::ForbiddenAccess
```
Comment (sluggoman) [inline lib/shopify_api/resources/usage_charge.rb]: why are you catching this - if you let it bubble up doesn't the client get a 403 and wouldn't that be preferred?

Comment (sluggoman) [inline lib/shopify_api/resources/usage_charge.rb]: Sorry - this is the client API already - this *is* the 403 coming back from the server

**Comment by ch33sybr3ad** [comment]
fixed https://github.com/Shopify/billing-test-app/blob/master/app/controllers/home_controller.rb#L4 @sluggoman

File: test/usage_charge_test.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+    assert_equal "1000 emails", usage_charge.description
+  end
+
+  def test_list_usage_charges
+    fake "recurring_application_charges/654381177/usage_charges", :method => :get, :status => 201, :body => load_fixture('usage_charges')
+
+    usage_charges = ShopifyAPI::UsageCharge.find(:all, params: {recurring_application_charge_id: 654381177})
+
+    assert_equal 2, usage_charges.length
```
Comment (sluggoman) [inline test/usage_charge_test.rb]: add a test for RecurringAppCharges with no usage charges

**Comment by sluggoman** [comment]
:+1:

**Comment by dominiquesr** [comment]
:+1:

**Pull request merged**